chef/automate
# [automate-2447] Preventing project deletion

### :nut_and_bolt: Description: What code changed, and why?

From the API, prevent the deletion of a project if it has rules associated with it. There are two cases to prevent project deletion. One, if the project has any rules associated with it. Second, if the project is in the edits pending state. 

### :chains: Related Resources

https://github.com/chef/automate/issues/2447

### :athletic_shoe: How to Build and Test the Change
1. `build components/authz-service && start_all_services`
1. Go to the project page https://a2-dev.test/settings/projects
1. Create a project with a rule.
1. Ensure the project can not be deleted.
1. Run apply-rules. 
1. Again ensure the project can not be deleted.
1. Delete the rule from the project. 
1. Then one more time, ensure the project can not be deleted.
1. Run apply-rules again.
1. Then ensure the project can be deleted. 

### :white_check_mark: Checklist
